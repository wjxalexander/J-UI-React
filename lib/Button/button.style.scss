@import "../_helper.scss";

%default-style {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  color: $default_white;
  font-family: inherit;
  text-decoration: none;
  border-style: none;
  min-width: 80px;
  max-width: 280px;
  border-radius: 2px;
  //min-height: 72px;
  //height: auto;
  height: 32px;
  font-size: $default_fontsize;
  font-weight: $default_text_title_fontweight;
  cursor: pointer;
  outline: none;
}

.j-ui-button {
  &-default-wrapper {
    @extend %default-style;
    background-color: $default_color;
    border: #000;
    &:hover,
    &:active {
      background-color: $button_hover_color;
    }
  }
;

  &-custom-wrapper {
    @extend %default-style;

    border: 1px solid $custom-border-color;
    background-color: #fff;
    color: #000;

    &:hover,
    &:active {
      background-color: rgb(243, 242, 241);
      color: rgb(32, 31, 30);
    }
  }
;

  &-secondaryText {
    flex-grow: 1;
    font-weight: 400;
    font-size: 12px;
  }
}

.j-ui-button-icon {
  font-size: $default_fontsize;
  margin: 0 4px;
}

button:disabled {
  color: rgba(0, 0, 0, 0.25);
  background-color: #f5f5f5;
  border-color: #d9d9d9;
  cursor: not-allowed;
  &:hover{
    color: rgba(0, 0, 0, 0.25);
    background-color: #f5f5f5;
  }
}