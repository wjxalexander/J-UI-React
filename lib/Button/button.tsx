import * as React from 'react';
import {rootClass} from "../utils/classFactory";
import Icon, {IconProps} from "../Icon/icon";
import "./button.style.scss"
import cls from "classnames"

export interface buttonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  title?: string;
  buttonType: "default" | "custom";
  iconProps?: IconProps;
  secondaryText?: string;
}

const classGenerator = rootClass('j-ui-button');

const Button: React.FunctionComponent<buttonProps> = (props: buttonProps) => {
  const {buttonType,iconProps,secondaryText,...restProps} = props;
  return (
    <button {...restProps} className={classGenerator(`${buttonType}-wrapper`)}>
      {iconProps ?
        <Icon {...iconProps} className={cls('j-ui-button-icon', iconProps.className)}/> : null}
      <div className={classGenerator(`${buttonType}-title`)}>
        {props.title || "button"}
      </div>
      {secondaryText ?
        <div className={classGenerator(`${buttonType}-secondaryText`)}>{secondaryText}</div> : null}
    </button>)
};
 Button.defaultProps = {
   buttonType: 'default'
 }
export default Button;