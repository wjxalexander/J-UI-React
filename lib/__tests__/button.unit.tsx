import renderer from 'react-test-renderer'
import Button from "../button"
import React from 'react'; // !!!
describe("button test",()=>{
    it("is a div",()=>{
        const tree = renderer
            .create(<Button/>)
            .toJSON() //虚拟dom是个对象
        expect(tree).toMatchSnapshot()
    })
})