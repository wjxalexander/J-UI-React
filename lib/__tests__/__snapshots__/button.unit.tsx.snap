// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test is a div 1`] = `
<div>
  234
</div>
`;
